### PR TITLE
Show project deletion error correctly

### LINKED ISSUE #1544: [Bug]: Error message for project deletion appears in the modal only in the next deletion attempt
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

OpenDataHub core version (eg. `v1.6.0`)

### Version

1.6.0

### Current Behavior

When a user with Edit permissions on a DS project tries to delete it, the modal box gets closed without showing any error messages. If user opens the deletion modal a second time, the error is rendered before confirming the deletion action. 

GIF representing the scenario:

deletion-error-msg

### Expected Behavior

To either delete the project or disable the action

### Steps To Reproduce

1. create a DS project with user A
2. share Edit permission to user B
3. try to delete the DS project using user B
4. try again

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1544 

## Description

We should not close the modal when the error appears, instead, we should keep the modal open and show the error.

Screenshot 2023-07-24 at 2 27 09 PM

## How Has This Been Tested?

1. Login as an admin user and create a project
2. Give a regular user `edit` permission in project sharing, so he could see the project in the project list
3. Login as this regular user and try to delete the project
4. See if the modal keeps open and the error shows

## Test Impact

Add deletion error to storybook test.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by DaoDaoNoCode** [comment]
@Gkrumbach07 Can you check the test cases here?

File: frontend/src/__tests__/integration/pages/projects/ProjectView.stories.tsx [hunk 1]
```diff
@@ -27,6 +28,18 @@ export default {
         rest.get('/api/k8s/apis/project.openshift.io/v1/projects', (req, res, ctx) =>
           res(ctx.json(mockK8sResourceList([mockProjectK8sResource({})]))),
         ),
+        rest.delete(
```
Comment (Gkrumbach07) [inline frontend/src/__tests__/integration/pages/projects/ProjectView.stories.tsx:31]: is there a reason you add the 403 error mock to the default story export, and not just the delete project story?

It doesn't really effect things, just a nit

**Review by Gkrumbach07 — APPROVED** [review]
/LGTM

File: frontend/src/__tests__/integration/pages/projects/ProjectView.stories.tsx (hunk 1 above)
Comment (DaoDaoNoCode) [reply frontend/src/__tests__/integration/pages/projects/ProjectView.stories.tsx:31]: Because I thought it might be reused in other delete modal stories.

**Review by manaswinidas — APPROVED** [review]
/lgtm

The changes work fine for me
Screenshot 2023-07-25 at 11 00 46 PM

**Pull request merged**